ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -80,8 +80,8 @@ import           Ouroboros.Consensus.Storage.ChainDB (ChainDB,
                      InvalidBlockReason)
 import qualified Ouroboros.Consensus.Storage.ChainDB as ChainDB
 
-type Consensus (client :: Type -> Type -> (Type -> Type) -> Type -> Type) blk m =
-   client (Header blk) (Tip blk) m ChainSyncClientResult
+type Consensus (client :: Type -> Type -> Type -> (Type -> Type) -> Type -> Type) blk m =
+   client (Header blk) (Point (Header blk)) (Tip blk) m ChainSyncClientResult
```

Comment:
If you prefer, we can turn this into `Point blk` with some calls to `castPoint` in the right places.